### ISSUE TITLE
Support for software model

### ISSUE THREAD

**Issue by Shad0w1nk** [linkedin/pygradle#30]
In the native world, we are extensively exposed to the new software model (components container) and we did a quick implementation of a Python plugin following this model. The main advantage is to have multiple component in the same project. For example, it allows you to have a component which is a REST server and another one which is the API to talk with the server colocated inside the same project. What's the roadmap for this? As a milestone, managed type could be used for simplifying the current implementation? I could provide some help on this. Thanks for the input.

**Comment by ethankhall** [comment]
Agreed! The goal is to switch to the software model with the 2.0 release. We have something that was working as of a few months ago as a spike. It would have to be updated to match any functionality that's changed but is something on the roadmap. If you're interested I can add it to the repo and you can play with it (maybe fix the changes that have happened to the api since then).

The one big thing we want to do is wait until the software model is stable, it seems to have been pretty stable the past few released so this might be a good time to look at it again.

**Comment by Shad0w1nk** [comment]
That would be nice. The model is indeed more stable, however it would be wise to wait a bit more. Still, I would be interested to look and help fix what you already have.

**Comment by ethankhall** [comment]
Sweet, I'll try to get a branch out there this week with it and I'll let you know.

**Comment by ethankhall** [comment]
@Shad0w1nk I pushed up a new branch: component-model-spike it looks like 2 tests are failing but the others are passing.

**Comment by Shad0w1nk** [comment]
Thanks @ethankhall, I had a quick look and it is very well designed. I will have a closer look and post some suggestion, if any.

**Comment by ethankhall** [comment]
Cool let me know what you find. I would be interested to hear your ideas on having different components being able to have different dependencies and maybe they extend from a shared dependency 'configuration'. That way as people have components with different versions of python, they could express the dependencies that are needed for different versions in a nice way, then we could convert those out to be used by the setup.py/DistClass to make it work nicely with the rest of the python ecosystem.
